hail_search/queries/mito.py

```diff
@@ -253,13 +284,20 @@ def _apply_multi_sample_type_entry_filters(self, ht, family_idx_map):
         return ht.filter(ht.family_entries.any(hl.is_defined))
 
     @staticmethod
-    def _family_has_valid_sample_type_entries(ht, sample_type, sample_type_family_idx):
-        # Note: This logic does not sufficiently handle case 2 here https://docs.google.com/presentation/d/1hqDV8ulhviUcR5C4PtNUqkCLXKDsc6pccgFVlFmWUAU/edit?usp=sharing
-        # and will need to be changed to support it - https://github.com/broadinstitute/seqr/issues/4403
+    def _family_has_valid_quality(ht, sample_type, sample_type_family_idx):
         return (
             hl.is_defined(sample_type_family_idx) &
-            hl.is_defined(ht[sample_type.passes_quality_field][sample_type_family_idx]) &
-            hl.is_defined(ht[sample_type.passes_inheritance_field][sample_type_family_idx])
+            hl.is_defined(ht[sample_type.passes_quality_field][sample_type_family_idx])
+        )
+
+    @staticmethod
+    def _family_has_valid_inheritance(ht, sample_type, family_idx, other_sample_type_family_idx):
+        return hl.bind(
+            lambda other_sample_type_pass_samples: (
+                ht[sample_type.failed_family_sample_field][family_idx].all(other_sample_type_pass_samples.contains)
+            ), ht[sample_type.other_sample_type.family_entries_field][other_sample_type_family_idx].filter(
+                lambda s: ~ht[sample_type.other_sample_type.failed_family_sample_field][other_sample_type_family_idx].contains(s['sampleId'])
```

Comment:
I think this won't work since `family_entries_field` and `failed_family_sample_field` for a given sample type are not necessarily the same length. ` ht[sample_type.other_sample_type.failed_family_sample_field][other_sample_type_family_idx]` is a list of sample_ids of variable length - it only contains failed samples - so the [i] access won't work. I originally had `failed_family_sample_field` as an array of failing indices but switched it to an array of failing sample IDs because it was easier for me to come up with a solution to compare the same samples from different sample types. 

Maybe a better way to structure `failed_family_sample_field` is an array of booleans that matches the shape of `family_entries_field` where the value at each sample index is true or false for failed. It's just more complicated to compare samples individually that way - I'd need to reintroduce that map of families to samples to sample types to indices inside of their respective family_entries field.

Reply:
>  I'd need to reintroduce that map of families to samples to sample types to indices inside of their respective family_entries field.

if `failed_family_sample_field` is a list of booleans then you can easily get the list of passing sample ids from the main family_entries field without maintaining an entry map, and then your logic for using `other_sample_type_pass_samples` would not need to change
```
hl.enumerate(ht[family_entries_field][family_idx]).starmap(
  lambda sample_idx, sample: hl.or_missing(
    ~ht[failed_family_sample_field][family_idx][sample_idx]),
    sample['sampleId'],
  )
).filter(hl.is_defined)
```

Reply:
ok, I refactored the `passes_inheritance_field` to be a list of booleans corresponding to the samples in family entries, ^ and used this more performant code to get a list of passing sample IDs